Plugin/Magento/Checkout/CustomerData/Cart.php

```diff
@@ -130,13 +186,115 @@ private function getQuote()
+        }
+        return $this->quoteMaskedId;
+    }
+
+    /**
+     * Pre-fetch cart main thread
+     *
+     * @param Quote $quote
+     * @param array $customerData
+     * @return void
+     * @throws LocalizedException
+     */
+    private function preFetchCart(Quote $quote, array $customerData): void
+    {
+        if (!$this->featureSwitches->isEnabledPreFetchCartViaApi()) {
+            return;
+        }
+        $customerDataHash = $this->getCustomerDataHash($customerData);
+        try {
+            // makes bolt pre-fetch cart request if customer-data is not in cache
+            if (!$this->getCustomerDataFromCache($customerDataHash)) {
+                $this->preFetchCartBoltRequest($quote, $customerData);
+                $this->saveCustomerDataToCache($customerData, $customerDataHash);
+            }
+        } catch (\Exception $e) {
+            $this->bugsnag->notifyException($e);
         }
-        return $maskedId;
+    }
+    /**
+     * Make pre fetch cart data to Bolt
+     *
+     * @param Quote $quote
+     * @param array $customerData
+     * @return void
+     * @throws AlreadyExistsException
+     * @throws LocalizedException
+     * @throws \Zend_Http_Client_Exception
+     */
+    private function preFetchCartBoltRequest(Quote $quote, array $customerData): void
+    {
+
+        $apiKey = $this->configHelper->getApiKey($quote->getStoreId());
+        $publishKey = $this->configHelper->getPublishableKeyCheckout($quote->getStoreId());
+        if (!$apiKey || !$publishKey) {
+            throw new LocalizedException(
+                __('Bolt API Key or Publishable Key - Multi Step is not configured')
+            );
+        }
+        $requestData = $this->dataObjectFactory->create();
+        $requestData->setDynamicApiUrl(ApiHelper::API_PRE_FETCH_CART)
+            ->setApiKey($apiKey)
+            ->setStatusOnly(true)
+            ->setRequestMethod('POST')
+            ->setHeaders(
+                [
+                    'X-Publishable-Key' => $publishKey
+                ]
+            )->setApiData(
+                [
+                    'order_reference' => $this->getQuoteMaskedId((int)$quote->getId())
+                ]
+            );
+        $request = $this->apiHelper->buildRequest($requestData);
+        $this->apiHelper->sendRequest($request);
+    }
+
+    /**
+     * Returns uniq hash of customer data
+     *
+     * @param array $customerData
+     * @return string
+     */
+    private function getCustomerDataHash(array $customerData): string
+    {
+        return md5($this->serializer->serialize($customerData));
+    }
+
+    /**
+     * Save customer data to cache
+     *
+     * @param string $customerDataHash
+     * @param array $customerData
+     * @return bool
+     */
+    private function saveCustomerDataToCache(array $customerData, string $customerDataHash): bool
+    {
+        return $this->cache->save(
+            $this->serializer->serialize($customerData),
```

Comment:
It's not necessary to save all data in cache because we don't need it.
this->cache->save doesn't all empty payload pass just simple value, like true

Reply:
Replaced with simple `'true'`